lib/spack/spack/parser.py

```diff
@@ -432,6 +447,41 @@ def parse(self, initial_spec: spack.spec.Spec) -> spack.spec.Spec:
         return initial_spec
 
 
+class EdgeAttributeParser:
+    __slots__ = "ctx", "literal_str"
+
+    def __init__(self, ctx, literal_str):
+        self.ctx = ctx
+        self.literal_str = literal_str
+
+    def parse(self):
+        attributes = {}
+        while True:
+            if self.ctx.accept(TokenType.KEY_VALUE_PAIR):
+                name, value = self.ctx.current_token.value.split("=", maxsplit=1)
+                name = name.strip("'\" ")
+                value = value.strip("'\" ").split(",")
+                attributes[name] = value
+                if name not in ("deptypes", "virtuals"):
```

Comment:
> renaming them will change the spec format, though, right?

Yes, since in the JSON representation these are basically reserved edge variant names.

> Is it worth implementing that at this point before this becomes cemented?

I would say no, but in any case I would do it in a separate PR (to come shortly after) so we can keep the two modifications (feature and renaming) separated.

> We could implement type and virtual as aliases and reserve type, deptypes, virtual, and virtuals, w/o changing the schema. These are unlikely to be used for anything else.

Yes, this could be done.